tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -288,25 +291,103 @@ private JsonNode getCombinedStatus() {
     }
 
     /**
-     * Create a task.
-     *
-     * @param id                    The ID of the task to create.
-     * @param spec                  The specification of the task to create.
+     * Creates and schedules a task.
      */
-    public void createTask(final String id, TaskSpec spec)
+    public void createAndScheduleTask(TaskDetail task)
             throws Throwable {
+        ManagedTask managedTask = createTask(task);
+
+        if (managedTask != null) {
+            long delayMs = scheduleTask(managedTask);
+            log.info("Created a new task {} with spec {}, scheduled to start {} ms from now.", task.id, task.spec, delayMs);
+        }
+    }
+
+    /**
+     * Creates and schedules the given tasks.
+     *
+     * @throws RequestConflictException - if a task with that ID and a different spec already exists
+     */
+    public void createAndScheduleTasks(List<TaskDetail> tasks) throws RequestConflictException, InvalidRequestException {
+        List<ManagedTask> managedTasks = new ArrayList<>();
+        if (tasks.stream().map(task -> task.id).collect(Collectors.toSet()).size() < tasks.size())
+            throw new RequestConflictException("Duplicate task IDs given");
+
+        for (TaskDetail task: tasks) {
+            try {
+                ManagedTask createdTask = createTask(task);
+                if (createdTask != null)
+                    managedTasks.add(createdTask);
+            } catch (RequestConflictException | InvalidRequestException e) {
+                throw e;
```

Comment:
I thought I left a comment about this earlier.  Anyway, we don't want to leave things in a partial state if one things fails.  If something fails, the task is in DONE state with an error, and every other task goes on as usual.

Reply:
Hmm.  This still has the problem where things can be partially applied, because we have a bunch of `CreateTask` runnables being processed separately.  It would be easier to just have a single `CreateTasks` runnable and pass it the map.  Then the whole thing could fail with a `RequestConflictException` if any task had a conflict.